streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/PublishSubjectImpl.kt

```diff
@@ -60,7 +64,9 @@ open class PublishSubjectImpl<T> : PublishSubject<T> {
     override fun subscribe(s: Subscriber<in T>) {
         val subscription = PublisherSubscription(s, onPublisherSubscriptionCancelled)
         s.onSubscribe(subscription)
-        addSubscription(subscription)
+        serialQueue.dispatch {
```

Comment:
What are the impacts of adding the new subscription on the serialQueue here?

Reply:
As far as I know none for now.

Reply:
I mean, why do we need this change if it doesn't have any impact?

Reply:
Well I thought you meant negative impacts, it does have a positive impact in the retry resubscription of a cold publisher as it properly orders the events